File: java/src/main/java/vsdk/toolkit/render/sgl/PolygonVertex.java
package vsdk.toolkit.render.sgl;

public class PolygonVertex {
    public double sx; // Screen space position (sometimes homo)
    public double sy;
    public double sz;
    public double sw;
    public double x; // World space position
    public double y;
    public double z;
    public double u; // Texture position (sometimes homogeneous)
    public double v;
    public double r; // (red,green,blue) color
    public double g;
    public double b;

    public double getCoord(int i) {
        switch (i) {
            case 0:
                return sx;
            case 1:
                return sy;
            case 2:
                return sz;
            case 3:
                return sw;
            case 4:
                return x;
            case 5:
                return y;
            case 6:
                return z;
            case 7:
                return u;
            case 8:
                return v;
            case 9:
                return r;
            case 10:
                return g;
            case 11:
                return b;
            default:
                return 0.0;
        }
    }
}

File: typescript/base/src/main/vsdk/toolkit/render/sgl/PolygonVertex.ts
export class PolygonVertex {
  public sx: number;
  public sy: number;
  public sz: number;
  public sw: number;
  public x: number;
  public y: number;
  public z: number;
  public u: number;
  public v: number;
  public r: number;
  public g: number;
  public b: number;

  public constructor() {
    this.sx = 0.0;
    this.sy = 0.0;
    this.sz = 0.0;
    this.sw = 0.0;
    this.x = 0.0;
    this.y = 0.0;
    this.z = 0.0;
    this.u = 0.0;
    this.v = 0.0;
    this.r = 0.0;
    this.g = 0.0;
    this.b = 0.0;
  }

  public getCoord(i: number): number {
    switch (i) {
      case 0:
        return this.sx;
      case 1:
        return this.sy;
      case 2:
        return this.sz;
      case 3:
        return this.sw;
      case 4:
        return this.x;
      case 5:
        return this.y;
      case 6:
        return this.z;
      case 7:
        return this.u;
      case 8:
        return this.v;
      case 9:
        return this.r;
      case 10:
        return this.g;
      case 11:
        return this.b;
      default:
        return 0.0;
    }
  }
}
